src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenStringConcat.cs

```diff
@@ -913,185 +913,164 @@ .maxstack  4
-  IL_00cb:  ldloca.s   V_2
-  IL_00cd:  constrained. ""bool""
-  IL_00d3:  callvirt   ""string object.ToString()""
-  IL_00d8:  call       ""string string.Concat(string, string, string)""
-  IL_00dd:  call       ""void System.Console.WriteLine(string)""
-  IL_00e2:  ldloca.s   V_0
-  IL_00e4:  constrained. ""char""
-  IL_00ea:  callvirt   ""string object.ToString()""
-  IL_00ef:  ldstr      ""10""
-  IL_00f4:  ldloca.s   V_1
-  IL_00f6:  constrained. ""char""
-  IL_00fc:  callvirt   ""string object.ToString()""
-  IL_0101:  ldstr      ""11""
-  IL_0106:  call       ""string string.Concat(string, string, string, string)""
-  IL_010b:  call       ""void System.Console.WriteLine(string)""
-  IL_0110:  ldstr      ""12""
-  IL_0115:  ldloca.s   V_0
-  IL_0117:  constrained. ""char""
-  IL_011d:  callvirt   ""string object.ToString()""
-  IL_0122:  ldstr      ""13""
-  IL_0127:  ldloca.s   V_1
-  IL_0129:  constrained. ""char""
-  IL_012f:  callvirt   ""string object.ToString()""
-  IL_0134:  call       ""string string.Concat(string, string, string, string)""
-  IL_0139:  call       ""void System.Console.WriteLine(string)""
-  IL_013e:  ldstr      ""a14b15a16""
-  IL_0143:  call       ""void System.Console.WriteLine(string)""
-  IL_0148:  ldc.i4.6
-  IL_0149:  newarr     ""string""
-  IL_014e:  dup
-  IL_014f:  ldc.i4.0
-  IL_0150:  ldloca.s   V_0
-  IL_0152:  constrained. ""char""
-  IL_0158:  callvirt   ""string object.ToString()""
-  IL_015d:  stelem.ref
-  IL_015e:  dup
-  IL_015f:  ldc.i4.1
-  IL_0160:  ldstr      ""17""
-  IL_0165:  stelem.ref
-  IL_0166:  dup
-  IL_0167:  ldc.i4.2
-  IL_0168:  ldloca.s   V_1
-  IL_016a:  constrained. ""char""
-  IL_0170:  callvirt   ""string object.ToString()""
-  IL_0175:  stelem.ref
-  IL_0176:  dup
-  IL_0177:  ldc.i4.3
-  IL_0178:  ldstr      ""18""
+  IL_001c:  call       ""string char.ToString()""
+  IL_0021:  ldstr      ""3""
+  IL_0026:  call       ""string string.Concat(string, string)""
+  IL_002b:  call       ""void System.Console.WriteLine(string)""
+  IL_0030:  ldstr      ""4""
+  IL_0035:  ldloca.s   V_1
+  IL_0037:  call       ""string char.ToString()""
+  IL_003c:  call       ""string string.Concat(string, string)""
+  IL_0041:  call       ""void System.Console.WriteLine(string)""
+  IL_0046:  ldc.i4.1
+  IL_0047:  stloc.2
+  IL_0048:  ldloca.s   V_2
+  IL_004a:  call       ""string bool.ToString()""
+  IL_004f:  ldstr      ""5""
+  IL_0054:  ldloca.s   V_0
+  IL_0056:  call       ""string char.ToString()""
+  IL_005b:  call       ""string string.Concat(string, string, string)""
+  IL_0060:  call       ""void System.Console.WriteLine(string)""
+  IL_0065:  ldstr      ""6""
+  IL_006a:  ldloca.s   V_1
+  IL_006c:  call       ""string char.ToString()""
+  IL_0071:  ldc.i4.7
+  IL_0072:  call       ""System.IntPtr System.IntPtr.op_Explicit(int)""
+  IL_0077:  stloc.3
+  IL_0078:  ldloca.s   V_3
+  IL_007a:  constrained. ""System.IntPtr""
+  IL_0080:  callvirt   ""string object.ToString()""
```

Comment:
why isn't this one using `call ""string IntPtr.ToString()"": https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/IntPtr.cs#L138